enclave/core/sgx/td.c

```diff
@@ -302,9 +305,15 @@ void td_clear(td_t* td)
     if (td->depth != 0 || td->callsites != NULL)
         oe_abort();
 
+    /* Save the stack guard before cleanup. */
+    uint64_t stack_guard = td->base.stack_guard;
+
     /* Clear base structure */
     memset(&td->base, 0, sizeof(td->base));
 
+    /* Restore the stack guard after cleanup. */
+    td->base.stack_guard = stack_guard;
```

Comment:
The relative sequence is as following:
td_init()
_handle_ecall()
    td_clear()
Since td_init() and td_clear() are not int the same level, the stack_guard need to be there to avoid  _handle_ecall's failure . Inspired by your suggestion, moving td_init() into handle_ecall() may make the calling sequence more reasonable.

Reply:
td_init() is moved into handle_ecall(). Now handle_ecall() can keep the consistency of td and this ugly lines are removed.